lib/IRGen/GenDecl.cpp

```diff
@@ -4374,6 +4374,99 @@ void IRGenModule::emitAccessibleFunctions() {
   }
 }
 
+void IRGenModule::emitRuntimeDiscoverableAttributes(
+    TinyPtrVector<FileUnit *> &filesToEmit) {
+  StringRef sectionName;
+  switch (TargetInfo.OutputObjectFormat) {
+  case llvm::Triple::DXContainer:
+  case llvm::Triple::GOFF:
+  case llvm::Triple::SPIRV:
+  case llvm::Triple::UnknownObjectFormat:
+    llvm_unreachable("Don't know how to emit attribute section for "
+                     "the selected object format.");
+  case llvm::Triple::MachO:
+    sectionName = "__TEXT, __swift5_rattrs, regular";
```

Comment:
@Azoy I didn't register new section in Swift-COFF, ELF and it would also require adding an entry to `llvm/BinaryFormat/Swift.def` would you like me to do that?